### ISSUE TITLE
Automatic Age out and Delete of a offline device

### ISSUE THREAD

**Issue by bmfmancini** [Cacti/cacti#2691]
Hey Everyone 

Just a thought   about building an option to automatically delete a device if it has been offline for a given amount of time 

Some places will use  orchestration systems to hand out management IP's for devices 
usually, there is a pool of IP's and when a device is added the IP is taken from the pool and when it's removed or decommissioned its added back to the pool this could be something like static DHCP or a generated template that is pushed to the device.

Now if that device is decommissioned  and the record is not deleted from Cacti then if that IP is ever re-used then the device will show up but for a totally different device
Since we use AUTOM8 to scan these IP subnets if that offline device was deleted and a new device was added we would pick up the new device and have it added to cacti 

On large deployments, it can be easy to miss a site down for a while so this will add more automation to keep the system clean and accurate.

The way I picture the feature is this 

Device offline say 1-month 
 notification is sent to the admin of devices down for 1 month or the chosen time frame
if no action is taken after 1 more week then the device is removed from cacti and the graphs could be saved if by some chance it was missed 

As far as I am aware this feature does not exist but if it does please let me know!

As always thank you for your work and effort!

**Comment by cigamit** [comment]
This would have to be a plugin, as its not a very common use case.  Ideally the orchestration engine that is decommissioning the device should also call into Cacti and remove the device during the decommission.  Then you don't have to worry about deleting anything.

Also, saving the graphs, and then putting them back in is not a simple affair.

**Comment by bmfmancini** [comment]
Yea your right more of a plugin case. 

For the call in do you mean via the PHP cli script?

**Comment by cigamit** [comment]
Yes, eventually we will have a Restful API, but we are not there yet.
